### PR TITLE
feat: new page for context menus

### PR DESCRIPTION
**Pull request by ElviraMustafina**
- new page for context menus
- new screenshots
- minor updates to code samples: added titles for windows, updated comments
- added info about localization

### CODE REVIEW

File: topics/compose/compose-desktop-context-menus.md [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+If you want to use a specific language, specify it as a default language explicitly before running your application: 
+
+```Console
+java.util.Locale.setDefault(java.util.Locale("en"))
+```
+
+## Swing interoperability
+
+If you are embedding Compose code into an existing Swing application and need the context menu to match the appearance and 
+behavior of other parts of the application, you can use the `JPopupTextMenu` class. In this class, `TextContextMenu` uses
```
Comment (igordmn) [inline topics/compose/compose-desktop-context-menus.md:307]:
```suggestion
behavior of other parts of the application, you can use the `JPopupTextMenu` class. In this class, `LocalTextContextMenu` uses 
```

File: topics/compose/compose-desktop-context-menus.md [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+<img src="compose-desktop-context-menu-dark-mode.png" alt="Context menu: Dark theme" width="500"/>
+
+## Override default context menu
+
+To override the default context menu for all text fields and selectable text elements, override the `TextContextMenu` interface.
+In the following code sample, we reuse the original `TextContextMenu`, but add one additional item to the bottom of the list.
+The new item adjusts to the text selection:
+
+```kotlin
+import androidx.compose.foundation.ContextMenuDataProvider
```
[the remark this answers is not in the document]
Comment (ElviraMustafina) [reply topics/compose/compose-desktop-context-menus.md:206]: Applying the patch somehow failed, so I updated the code sample manually.

File: topics/compose/compose-desktop-context-menus.md [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+{initial-collapse-state="collapsed" collapsible="true" collapsed-title="ContextMenuDataProvider( items = { listOf( ContextMenuItem( "}
+
+<img src="compose-desktop-context-menu-custom-actions.png" alt="Context menu with custom actions" width="500"/>
+
+If you want to override the text menu for all text fields and selectable texts in your application, refer to the section about [`TextContextMenu`](#override-default-context-menu).
+
+## Context menu in a custom area
+
+You can create a context menu for any arbitrary area of your application. Use `ContextMenuArea` to define a container where
+the right mouse click will trigger the appearance of a context menu:
```
Comment (zamulla) [inline topics/compose/compose-desktop-context-menus.md:114]: The intro to this page talks about support for text context menus, and I'm under the impression that this is all Compose can do for desktop, but here we can create a context menu for anything. First, I would make it clearer in the intro, that we can have context menus everywhere; second, maybe put more abstract examples first (this one and theming), and put everything else in a separate section dedicated to text? Right now everything is on the same level of header structure and may be disorienting.

**Review by zamulla — CHANGES_REQUESTED** [review]
Nice! There are small suggestions for text, but the most important one is about structure: separating the text context menus from everything else.

File: topics/compose/compose-desktop-context-menus.md [hunk 4]
```diff
@@ -420,6 +408,18 @@ private fun circleIcon(color: Color) = object : Icon {
 
 <img src="compose-desktop-context-menu-swing.png" alt="Context menu: Swing interoperability" width="500"/>
 
+## Localized menu items
```
Comment (zamulla) [inline topics/compose/compose-desktop-context-menus.md:411]: Should this be moved to the beginning as well, before we get to text context menus?

**Pull request merged**